base/linalg/dense.jl

```diff
@@ -291,7 +291,7 @@ julia> kron(A, B)
 ```
 """
 function kron{T,S}(a::AbstractMatrix{T}, b::AbstractMatrix{S})
-    R = Array{promote_type(T,S)}(size(a,1)*size(b,1), size(a,2)*size(b,2))
+    R = Array{promote_op(*,T,S)}(size(a,1)*size(b,1), size(a,2)*size(b,2))
```

Comment:
I actually don't understand all the magic of comprehensions and why they can be better than `return_type`... but yes `promote_op` is better than `promote_type`. At least it is correct when inference works!